ibm/resource_ibm_network_gateway.go

```diff
@@ -0,0 +1,895 @@
+package ibm
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/softlayer/softlayer-go/datatypes"
+	"github.com/softlayer/softlayer-go/filter"
+	"github.com/softlayer/softlayer-go/helpers/location"
+	"github.com/softlayer/softlayer-go/helpers/product"
+	"github.com/softlayer/softlayer-go/services"
+	"github.com/softlayer/softlayer-go/session"
+	"github.com/softlayer/softlayer-go/sl"
+)
+
+const packageKeyName = "NETWORK_GATEWAY_APPLIANCE"
+
+func resourceIBMNetworkGateway() *schema.Resource {
+	return &schema.Resource{
+		Create:   resourceIBMNetworkGatewayCreate,
+		Read:     resourceIBMNetworkGatewayRead,
+		Update:   resourceIBMNetworkGatewayUpdate,
+		Delete:   resourceIBMNetworkGatewayDelete,
+		Exists:   resourceIBMNetworkGatewayExists,
+		Importer: &schema.ResourceImporter{},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"hostname": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				DefaultFunc: genID,
+				DiffSuppressFunc: func(k, o, n string, d *schema.ResourceData) bool {
+					// FIXME: Work around another bug in terraform.
+					// When a default function is used with an optional property,
+					// terraform will always execute it on apply, even when the property
+					// already has a value in the state for it. This causes a false diff.
+					// Making the property Computed:true does not make a difference.
+					if strings.HasPrefix(o, "terraformed-") && strings.HasPrefix(n, "terraformed-") {
+						return true
+					}
+					return o == n
+				},
+			},
+
+			"domain": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"user_metadata": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"notes": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"post_install_script_uri": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Default:          nil,
+				ForceNew:         true,
+				DiffSuppressFunc: applyOnce,
+			},
+
+			"tags": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+
+			"os_reference_code": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "OS_VYATTA_5600_5_X_UP_TO_1GBPS_SUBSCRIPTION_EDITION_64_BIT",
+				ForceNew: true,
+			},
+
+			"datacenter": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+
+			"network_speed": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  100,
+				ForceNew: true,
+			},
+			//Does not apply for Network gateway but is required for cancellation
+			"hourly_billing": {
```

Comment:
Why is it required during cancellation 🤔 ? Can't we simply assume it is monthly and use its value as false?

Reply:
This has been changed to reflect a default value of FALSE